app/ui.js

```diff
@@ -936,12 +938,43 @@ const UI = {
         }
     },
 
+    readClipboard(callback) {
+      var readfuture = navigator.clipboard.readText();
```

Comment:
This API isn't available everywhere, so we need some handling here when it isn't.